### PR TITLE
better-batchList-generating in removeObjects

### PR DESCRIPTION
**Pull request by KevinSmile**
better-batchList-generating in removeObjects

as discussed in https://github.com/minio/minio-java/issues/810

### CODE REVIEW

File: api/src/main/java/io/minio/MinioClient.java [hunk 1]
```diff
@@ -3009,24 +3010,27 @@ public void removeObject(String bucketName, String objectName)
           private Result<DeleteError> error;
           private Iterator<DeleteError> errorIterator;
           private boolean completed = false;
-          private Iterator<String> objectNameIter = objectNames.iterator();
+          private final int batchSize = 1000;
```
Comment (balamurugana) [inline api/src/main/java/io/minio/MinioClient.java]: you could hard-code `1000` i.e. no need of `batchSize` variable.

Comment (KevinSmile) [reply api/src/main/java/io/minio/MinioClient.java]: ok. done.

File: api/src/main/java/io/minio/MinioClient.java [hunk 2]
```diff
@@ -3009,24 +3010,26 @@ public void removeObject(String bucketName, String objectName)
           private Result<DeleteError> error;
           private Iterator<DeleteError> errorIterator;
           private boolean completed = false;
-          private Iterator<String> objectNameIter = objectNames.iterator();
+          // divide the objectNames to sublists of the given batchSize
+          private Iterator<List<String>> subObjectListsIter =
+                  Iterables.partition(objectNames, 1000).iterator();
```
Comment (balamurugana) [inline api/src/main/java/io/minio/MinioClient.java]: you could change the comment and variable name.
```java
// divide given objectNames into list of object names on which each list contains maximum of 1000 names.
private Iterator<List<String>>  batchedObjectNames = Iterables.partition(objectNames, 1000).iterator();
```

File: api/src/main/java/io/minio/MinioClient.java [hunk 3]
```diff
@@ -3009,24 +3010,26 @@ public void removeObject(String bucketName, String objectName)
...
-              if (i > 0) {
-                errorList = removeObject(bucketName, objectList);
+              if (subObjectListsIter.hasNext()) {
+                List<DeleteObject> subDeleteObjectList = subObjectListsIter.next().stream().map(objectName -> {
+                  try {
+                    return new DeleteObject(objectName);
+                  } catch (XmlPullParserException e) {
+                    throw new RuntimeException(e);
+                  }
+                }).collect(Collectors.toList());
```
Comment (balamurugana) [inline api/src/main/java/io/minio/MinioClient.java]: I don't know what advantages we get here. My choice is `for` loop which is quite simple (and to read) than java8 stream here.  Do we get any performance improvement here?

**Comment by balamurugana** [comment]
@KevinSmile You would need to check travis-ci failures and fix them accordingly.

**Comment by nitisht** [comment]
@KevinSmile closing this as stale, feel free to reopen once you have time to address the review comments.

**Pull request closed without merging**